bitcoin/src/blockdata/witness.rs

```diff
@@ -248,6 +248,13 @@ impl Witness {
         witness
     }
 
+    /// Creates a witness required to do a key path spend of a P2TR output.
+    pub fn p2tr_key_spend(signature: &taproot::Signature) -> Witness {
```

Comment:
Just realized we could mark signatures if they are key spend or script spend. It'd actually help in case of a recent issue #2380

For a future PR of course. And possibly with the limitation that we don't take into account which leaf script it is.